lib/spack/spack/traverse.py

```diff
@@ -18,7 +18,7 @@
 
 
 def sort_edges(edges):
-    edges.sort(key=lambda edge: edge.spec.name)
+    edges.sort(key=lambda edge: edge.spec.name or edge.spec.abstract_hash)
```

Comment:
You probably meant `(name or  '', abstract_hash or '')`, it's weird to order as `(bar, /baz, foo)`

Reply:
I agree w/this